src/client/entity-editor/alias-editor/actions.ts

```diff
@@ -87,7 +85,7 @@ export function debouncedUpdateAliasSortName(
  * @param {number} value - The new value to be used for the alias language ID.
  * @returns {Action} The resulting UPDATE_ALIAS_LANGUAGE action.
  */
-export function updateAliasLanguage(rowId: number, value: ?number): Action {
+export function updateAliasLanguage(rowId: number, value: number | null | undefined): Action {
```

Comment:
Flow and TypeScript both have the `param?: type` syntax for optional parameters. But most/all of these occurences of `<type> | null | undefined` are replacing `?<type>` rather than `param?`, which doesn't necessarily signify an optional parameter - just a [maybe type](https://flow.org/en/docs/types/maybe/). So I'm not sure we can say that all of them should be optional by default.

Taking `updateAliasLanguage` as an example, the only call to this function is in `src/client/entity-editor/alias-editor/alias-row.js`:

    function mapDispatchToProps(dispatch, {index}) {
            return {
                    onLanguageChange: (value) =>
                            dispatch(updateAliasLanguage(index, value && value.value)),
                    onNameChange: (event) =>
                            dispatch(debouncedUpdateAliasName(index, event.target.value)),
                    onPrimaryClick: (event) =>
                            dispatch(updateAliasPrimary(index, event.target.checked)),
                    onRemoveButtonClick: () =>
                            dispatch(removeAliasRow(index)),
                    onSortNameChange: (event) =>
                            dispatch(debouncedUpdateAliasSortName(index, event.target.value))
            };
    }

Since this is the only call, the 'value' parameter is never not supplied, so we can't really say that this should be optional. What's happening here is that value can be null (if the alias language is unselected). So we could maybe reduce the parameter type down to `number | null`, but I wouldn't be confident in doing that until we had better typing for the Select and LanguageField components to prove that this function can never receive `undefined`.

In summary, what I would say is this: this whole PR is a very minimal conversion from Flow to TypeScript, and there are a whole load of areas where we can improve typing and make things clearer. I think there'll be plenty of opportunities to do that once the basics are in. Until we have 100% type coverage and no `any` types anywhere, it's always a work-in-progress.